### PR TITLE
Move wsoc 2 bundle api to private feature

### LINKED ISSUE #15323: The Jakarta 9 Websocket 2.0 API needs a private feature
Currently the websocket-2.0 feature just pulls in the API bundle.

In addition the faces-3.0 feature also pulls in the WebSocket 2.0 API bundle.

We should make a private feature for the WebSocket 2.0 API , that also includes the EE9 compatibility feature and follows a similar structure to other Jakarta EE9 private API features.

The faces-3.0 feature and the websocket-2.0 feature should then include this new private feature rather than the API bundle.

### PR DESCRIPTION
**Pull request by isaacrivriv**
fixes #15323

### CODE REVIEW

**Comment by isaacrivriv** [comment]
#build

**Comment by isaacrivriv** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#run-libby-bot

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.websocket-2.0.feature [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.jakarta.websocket-2.0
+singleton=true
+-features=com.ibm.websphere.appserver.eeCompatible-9.0
+-bundles=io.openliberty.jakarta.websocket.2.0; location:="dev/api/spec/,lib/"; mavenCoordinates="jakarta.websocket:jakarta.websocket-api:2.0.0"
+kind=beta
+edition=core
+WLP-Activation-Type: parallel
```
Comment (pnicolucci) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.websocket-2.0.feature:8]: fixing the new line error would be ideal.

**Review by pnicolucci — COMMENTED** [review]
Just a minor comment.

**Comment by isaacrivriv** [comment]
#build

**Review by tevans78 — APPROVED** [review]

**Review by pnicolucci — APPROVED** [review]

**Comment by isaacrivriv** [comment]
#run-libbybot

**Pull request merged**